tests/turbo.node.test.ts

```diff
@@ -194,20 +195,128 @@ describe('Node environment', () => {
+          expect(response).to.have.property('owner');
+          expect(response['owner']).to.equal(jwkToPublicArweaveAddress(jwk));
         });
-        expect(response).to.not.be.undefined;
-        expect(response).to.not.be.undefined;
-        expect(response).to.have.property('fastFinalityIndexes');
-        expect(response).to.have.property('dataCaches');
-        expect(response).to.have.property('owner');
-        expect(response['owner']).to.equal(jwkToPublicArweaveAddress(jwk));
-      });
+      }
+
+      const invalidDataItemOpts = [
+        {
+          testName: 'tag name too long',
+          errorType: 'FailedRequestError',
+          errorMessage: 'Failed request: 400: Data item parsing error!',
+          dataItemOpts: {
+            tags: [
+              {
+                name: Array(1025).fill('a').join(''),
+                value: 'test',
+              },
+            ],
+          },
+        },
+        {
+          testName: 'tag value too long',
+          errorType: 'FailedRequestError',
+          errorMessage: 'Failed request: 400: Data item parsing error!',
+          dataItemOpts: {
+            tags: [
+              {
+                name: 'test',
+                value: Array(3073).fill('a').join(''),
+              },
+            ],
+          },
+        },
+        {
+          testName: 'target Too Short',
+          errorMessage: 'Target must be 32 bytes but was incorrectly 10',
+          dataItemOpts: {
+            target: 'target Too Short',
+          },
+        },
+        {
+          testName: 'anchor Too Short',
+          errorMessage: 'Anchor must be 32 bytes',
+          dataItemOpts: {
+            anchor: 'anchor Too Short',
+          },
+        },
+        {
+          testName: 'target Too Long',
+          errorMessage: 'Target must be 32 bytes but was incorrectly 33',
+          dataItemOpts: {
+            target: 'target Too Long This is 33 Bytes one two three four five',
+          },
+        },
+        {
+          testName: 'anchor Too Long',
+          errorMessage: 'Anchor must be 32 bytes',
+          dataItemOpts: {
+            anchor: 'anchor Too Long This is 33 Bytes one two three four five',
+          },
+        },
+      ];
+      for (const {
+        testName,
+        dataItemOpts,
+        errorMessage,
+        errorType,
+      } of invalidDataItemOpts) {
+        it(`should fail to upload a Buffer to turbo with invalid  when ${testName}`, async () => {
+          const filePath = new URL('files/1KB_file', import.meta.url).pathname;
+          const fileSize = fs.statSync(filePath).size;
+
+          await expectAsyncErrorThrow({
+            promiseToError: turbo.uploadFile({
+              fileStreamFactory: () => fs.createReadStream(filePath),
+              fileSizeFactory: () => fileSize,
+              dataItemOpts,
+            }),
+            errorType,
+            errorMessage,
+          });
```

Comment:
I've been using:

```
const error = await my promise.catch(e => e)
expect(error).toBeInstsnceOf(Error);
expect(error.message).to.equal('my error message');
```

It's fairly straightforward and avoids all wrapper stuff. Fine what you have / just wanted to share